design-proposals/live-migration-for-bridged-pod-network.md

```diff
@@ -0,0 +1,54 @@
+# Overview
+This feature enables support of live-migrate for VMs with pod network in bridge mode.
+
+## Motivation
+Masquerade and slirp are not so performant compared to bridge mode.
+We want to use the less latency modes with the opportunity to live-migrate the virtual machines.
+
+## Goals
+Provide a live-migration feature for VMs running with pod network connected in bridged mode.
+
+## Non Goals
+Live migration with network reconfiguration is not seamless, sometimes it can take some time for applying a new settings.
+
+## Definition of Users
+Everyone who use bridge to bind pod network may want to live-migrate created VMs.
+
+## User Stories
+* As a user / admin, I want to have an opportunity for live-migration of a VM with bridged pod-network.
+
+## Repos
+- [KubeVirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+
+To add a new feature gate `NetworkAwareLiveMigration`, which enables two additional methods in live-migration procedure:
```

Comment:
- I think the name better be specific here, e.g. `DisruptiveBridgeBindingMigration`.
- Could you please elaborate on when in the (migration) flow you plan to execute the detach/attach operation?
- Could you please elaborate on when in the (migration) flow you plan to execute the link down/up operation?
- If I understand correctly, the IP address is always expected to change, no matter if the mac is changed or not. Right?

Reply:
> I think the name better be specific here, e.g. DisruptiveBridgeBindingMigration.

I wouldn't specify any specific mode in feature gate name. The [next PR](https://github.com/kubevirt/kubevirt/pull/7648) is upcoming, it will bring macvtap binding mode which also will work with this feature.

> - Could you please elaborate on when in the (migration) flow you plan to execute the detach/attach operation?
> - Could you please elaborate on when in the (migration) flow you plan to execute the link down/up operation?

That was described here

https://github.com/kubevirt/community/blob/eb0d5dddf8aac2ba5e817e5a79b5ef9e7aa54812/design-proposals/live-migration-for-bridged-pod-network.md?plain=1#L26-L32

If you talking about when exactly is it going to happen, sure, I will add the information that this will happened after the successful live-migration


> If I understand correctly, the IP address is always expected to change, no matter if the mac is changed or not. Right?
Not exactly, some CNIs allow to specify IP-address in annotation, this way IP-address will not be changed, but we still need to link down and link up the interface to make sure that VM has correct routes.

Reply:
What's the criteria for this feature to graduate from a feature flag to being default enabled?

Reply:
>What's the criteria for this feature to graduate from a feature flag to being default enabled?

it would need to function for the majority of guest OSs and not cause undesirable behavior for existing installs. I'm unsure if this will reach a point where we can make it on by default, but it's a possibility.